holoviews/core/dimension.py

```diff
@@ -639,7 +639,11 @@ def map(self, map_fn, specs=None, clone=True):
             for k, v in self.items():
                 new_val = v.map(map_fn, specs, clone)
                 if new_val is not None:
-                    deep_mapped[k] = new_val
+                    # Ensure key validation doesn't cause errors
```

Comment:
I don't quite get why there would be key errors: ``deep_mapped`` is a clone of ``self`` and ``k`` comes from ``self.items()`` so why would the key ever be rejected?

Reply:
Just had to double check to figure out what's going on, apparently this can occur when working with multiple DynamicMaps in a Layout. I've not been able to track down the actual bug but it seems something else is bypassing the validation so you end up with a cache that contains values it shouldn't. I'll try to track down the original bug that leads to the invalid key being inserted in the first place.

Reply:
Found it, DynamicMap is inserting stuff into the cache with checking first.